felix/wireguard/wireguard.go

```diff
@@ -1389,21 +1408,22 @@ func (w *Wireguard) constructWireguardDeltaForResync(wireguardClient netlinkshim
 
 // ensureLink checks that the wireguard link is configured correctly. Returns true if the link is oper up.
 func (w *Wireguard) ensureLink(netlinkClient netlinkshim.Interface) (bool, error) {
-	logCxt := log.WithField("ifaceName", w.config.InterfaceName)
+	logCtx := w.logCtx.WithField("ifaceName", w.interfaceName)
 
-	if w.config.EncryptHostTraffic {
-		log.Debug("Enabling src valid mark for WireGuard")
+	if w.config.EncryptHostTraffic && w.ipVersion == 4 {
+		//TODO: what is the IPv6 equivalent for this?
```

Comment:
https://docs.kernel.org/networking/ip-sysctl.html

> IPv6 has no global variables such as tcp_*. tcp_* settings under ipv4/ also apply to IPv6 [XXX?].

This seems to suggest that perhaps the v4 setting will also work for v6? Seems a bit odd, and not very clear from the documentation... Might need to try it out. Or we might need to find an alternative way to get rpfilter to be happy with our fwmark shenanigans.